packages/apollo-link-http/src/httpLink.ts

```diff
@@ -179,73 +302,66 @@ export const createHttpLink = (linkOptions: HttpLink.Options = {}) => {
   if (!fetcher) fetcher = fetch;
   if (!uri) uri = '/graphql';
 
-  return new ApolloLink(
-    operation =>
-      new Observable(observer => {
-        const {
-          headers,
-          credentials,
-          fetchOptions = {},
-          uri: contextURI,
-          http: httpOptions = {},
-        } = operation.getContext();
-        const { operationName, extensions, variables, query } = operation;
-        const http = { ...defaultHttpOptions, ...httpOptions };
-        const body = { operationName, variables };
-
-        if (includeExtensions || http.includeExtensions)
-          (body as any).extensions = extensions;
-
-        // not sending the query (i.e persisted queries)
-        if (http.includeQuery) (body as any).query = print(query);
-
-        let serializedBody;
-        try {
-          serializedBody = JSON.stringify(body);
-        } catch (e) {
-          const parseError = new Error(
-            `Network request failed. Payload is not serializable: ${e.message}`,
-          ) as ClientParseError;
-          parseError.parseError = e;
-          throw parseError;
+  //currently all operations in a single batch use the same uri
+  let choosenURI: string;
+
+  return new BatchLink({
```

Comment:
Hmmm, I think I would have been expecting something that doesn't go through the batching code path at all unless batching is being used. This looks like it depends on what is passed into the BatchLink constructor, but maybe we could just avoid calling that?

I could imagine a situation where you do something like

```
ApolloLink.from([
  batchLink,
  httpLink
]);
```

or maybe that's wishful thinking

Reply:
Yeah @jaydenseric had similar sentiments! I think that `batchLink.concat(httpLink)` would complicate the request interface and make visual debugging(dev tools tracing a request through the chain) more difficult.

Whether we add the batching behavior to link-http or keep it in link-batch-http is a philosophical question to answer. I'm fine with having the terminating(transport) links owning batching, since I hope that batching becomes a standard. Then a developer will only have to pick their protocol: ws, ddp, http, etc. Adding and maintaining a separate batch version seems like a pain.